### PR TITLE
Make MSan builds with instrumented libraries (#608)

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

**Comment by oliverchang** [comment]
TBR since this is fairly small and I want to get some early results.

File: infra/base-images/base-builder/compile [hunk 1]
```diff
@@ -29,9 +29,17 @@ fi
...
+  # Take all libraries from lib/msan and MSAN_LIBS_PATH
   # export CXXFLAGS_EXTRA="-L/usr/msan/lib $CXXFLAGS_EXTRA"
   cp -R /usr/msan/lib/* /usr/lib/
+
+  if [[ -z "${MSAN_LIBS_PATH-}" ]]; then
+    echo 'WARNING: Building without MSan instrumented libraries.'
+  else
+    # Copy all static libraries only. Don't include .so files because they can
+    # break non MSan compiled programs.
+    (cd "$MSAN_LIBS_PATH" && find . -name '*.a' -exec cp --parents '{}' / ';')
```
Comment (Dor1s) [inline infra/base-images/base-builder/compile]: where do we copy these to?

Comment (oliverchang) [reply infra/base-images/base-builder/compile]: To /. These libraries are e.g. /usr/lib/libz.a etc

Comment (Dor1s) [reply infra/base-images/base-builder/compile]: Ah, so there is a relative path! Thanks.

**Pull request merged**